### PR TITLE
add counter-strike 2 log parser

### PR DESCRIPTION
**Pull request by boyski33**
I removed the `examples` folder, as I can't share a log file just yet, but maybe in the near future. I also updated the `.gitignore`.

### CODE REVIEW

**Comment by janstuemmel** [comment]
i'll look into it till end of the week

File: .gitignore [hunk 1]
```diff
@@ -1 +1,23 @@
+# Compiled Object files, Static and Dynamic libs (Shared Objects)
```
Comment (janstuemmel) [inline .gitignore:1]: We don't need those c specific things i guess...

File: .gitignore [hunk 2]
```diff
@@ -1 +1,23 @@
...
+*.o
+*.a
+*.so
```
Comment (janstuemmel) [inline .gitignore]: We don't need those c specific things i guess...

File: README.md [hunk 3]
```diff
@@ -1,60 +1,64 @@
...
+
 ```json
 {
   "time": "2018-11-05T15:44:36Z",
   "type": "PlayerPurchase",
   "player": {
     "name": "Player",
     "id": 12,
-    "steam_id": "STEAM_1:1:0101011",
+    "steam_id": "[U:1:29384012]",
```
Comment (janstuemmel) [inline README.md:61]: Are these the new steam ids?

File: cs2log.go [hunk 4]
```diff
@@ -360,49 +356,49 @@ const (
 	// TeamNoticePattern regular expression
 	TeamNoticePattern = `Team "(CT|TERRORIST)" triggered "(\w+)" \(CT "(\d+)"\) \(T "(\d+)"\)`
 	// PlayerConnectedPattern regular expression
-	PlayerConnectedPattern = `"(.+)<(\d+)><([\w:]+)><>" connected, address "(.*)"`
+	PlayerConnectedPattern = `"(.+)<(\d+)><([\[\]\w:]+)><>" connected, address "(.*)"`
```
Comment (janstuemmel) [inline cs2log.go:359]: is this correct? Shouldn't it be `\[\w+\]`. Can you provide a example line?

File: example/main.go [hunk 5]
```diff
@@ -1,62 +0,0 @@
-package main
```
Comment (janstuemmel) [inline example/main.go]: Shouldn't we keep the example? I guess the example.log can be deleted, but we can keep the example program

File: go.mod [hunk 6]
```diff
@@ -1,3 +1,3 @@
 module github.com/janstuemmel/csgo-log
 
-go 1.13
+go 1.19
```
Comment (janstuemmel) [inline go.mod:3]: generics support :)

File: README.md (hunk 3 above)
Comment (boyski33) [reply README.md:61]: Yeah, these are in the Steam ID 3 format.

File: .gitignore (hunk 2 above)
Comment (boyski33) [reply .gitignore]: Maybe not. I got a "standard Go gitignore". I'll remove them.

File: cs2log.go (hunk 4 above)
Comment (boyski33) [reply cs2log.go:359]: It doesn't matter, as it's in a group, so this just lists the allowed characters, so either way works. Here's a example line you can test it against:

`L 05/18/2023 - 14:10:10: "user<1><[U:1:2828281]><CT>" [-412 1946 -127] killed "Johnny<11><BOT><TERRORIST>" [-405 -132 -0] with "awp"`

File: example/main.go (hunk 5 above)
Comment (boyski33) [reply example/main.go]: Sure.

File: cs2log.go (hunk 4 above)
Comment (janstuemmel) [reply cs2log.go:359]: i'll try tomorrow, i guess it's time to merge then...

Comment (boyski33) [reply cs2log.go:359]: Yeah, try that, and feel free to merge my PR.

**Review by janstuemmel — APPROVED** [review]

**Pull request merged**